index.js

```diff
@@ -407,6 +407,72 @@ module.exports = {
         if (err) return cb(clarify(err, 'Error finding or creating additions feed when starting ssb-tribes2'))
         return cb()
       })
+
+      // look for new epochs that we're added to
+      ssb.metafeeds.findOrCreate((err, myRoot) => {
+        // prettier-ignore
+        if (err) return cb(clarify(err, 'Error getting own root in start()'))
+
+        pull(
+          ssb.db.query(
+            // TODO: does this output new stuff if we accept an invite to an old epoch and then find additions to newer epochs?
+            where(and(isDecrypted('box2'), type('group/add-member'))),
+            live({ old: true }),
+            toPullStream()
+          ),
+          // groups/epochs we're added to
+          pull.filter((msg) => {
+            return msg.value?.content?.recps?.includes(myRoot.id)
+          }),
+          // to find new epochs we only check groups we've accepted the invite to
+          paraMap((msg, cb) => {
+            pull(
+              ssb.box2.listGroupIds(),
+              pull.collect((err, groupIds) => {
+                // prettier-ignore
+                if (err) return cb(clarify(err, "Error getting groups we're already in when looking for new epochs"))
+
+                if (groupIds.includes(msg.value?.content?.recps?.[0])) {
+                  return cb(null, msg)
+                } else {
+                  return cb()
+                }
+              })
+            )
```

Comment:
we can move faster by comparing as we go and stopping as soon as we find a match

```suggestion
            const groupId = msg.value?.content?.recps?.[0]
            if (!groupId) return cb(null, null)

            pull(
              ssb.box2.listGroupIds(),
              pull.filter(id => id === groupId),
              pull.take(1),
              pull.collect((err, groupIds) => {
                // prettier-ignore
                if (err) return cb(clarify(err, "Error getting groups we're already in when looking for new epochs"))

                if (groupIds.length) return cb(null, msg)
                else cb(null, null)
              })
            )
```

Reply:
listGroupIds is basically sync, it's just a stream to allow for its live opt